sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpAsyncResponse.java

```diff
@@ -111,6 +116,71 @@ public void writeBodyTo(OutputStream outputStream) throws IOException {
         StreamUtils.INSTANCE.transfer(responseBody.byteStream(), outputStream);
     }
 
+    @Override
+    public Mono<Void> writeBodyTo(AsynchronousFileChannel asynchronousFileChannel, long position) {
+        return Mono.defer(
+            () -> {
+                AtomicLong currentPosition = new AtomicLong(position);
+                byte[] buffer = new byte[8 * 1024];
+                ByteBuffer byteBuffer = ByteBuffer.wrap(buffer);
+                BufferedSource source = responseBody.source();
+                try {
+                    int read = source.read(buffer);
+                    byteBuffer.position(0);
+                    byteBuffer.limit(read);
+
+                    // TODO (kasobol-msft) consider using just reactor here.
+                    CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+
+                    writeToChannel(
+                        asynchronousFileChannel, currentPosition, buffer, byteBuffer, source, completableFuture);
+
+                    return Mono.fromFuture(completableFuture);
+                } catch (IOException e) {
+                    return Mono.error(e);
+                }
+            }
+        ).doFinally(ignored -> close());
+    }
+
+    private void writeToChannel(
+        AsynchronousFileChannel asynchronousFileChannel, AtomicLong currentPosition,
+        byte[] buffer, ByteBuffer byteBuffer, BufferedSource source, CompletableFuture<Void> completableFuture) {
+        asynchronousFileChannel.write(byteBuffer, currentPosition.get(), byteBuffer,
+            new CompletionHandler<Integer, ByteBuffer>() {
+                @Override
+                public void completed(Integer result, ByteBuffer attachment) {
+                    currentPosition.addAndGet(result);
+
+                    if (byteBuffer.hasRemaining()) {
+                        // If the entire ByteBuffer hasn't been written send it to be written again until it completes.
+                        writeToChannel(
```

Comment:
Could this recursion result in a StackOverflowError if the response body is large enough?